.github/workflows/docs-preview-build.yml

```diff
@@ -0,0 +1,49 @@
+
+# This workflow builds the docs for each PR (including from forks), and uploads
+# the rendered docs as an artifact.
+#
+# The artifact will be picked up by docs-preview-deploy.yml and uploaded to
+# https://jj-preview.github.io/docs/.
+#
+# The split into "build" and "deploy" follows GitHub's guide at
+# https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/.
+# The build workflow has no access to the repo secrets. The deploy workflow runs
+# in the context of the main repo rather than the PR branch.
```

Comment:
Thanks, I learned a lot about GitHub actions from this PR.

Reply:
tbh after making this PR I am motivated to learn about other CIs. I used to think they were all roughly the same nowadays, and the biggest annoyance was yaml; now I think yaml is the smallest annoyance. I wish for a feature-complete CI 🥲

Reply:
Somewhere in Docusaurus docs, they recommend using Netlify or Vercel instead of GitHub pages since (they say) it's easier to do PR previews with them.

No idea whether it's better now that you've done the work. Also, we probably should wait to have a `jj` domain before jumping around providers, so that it doesn't affect URLs too much.

Reply:
Netlify and Vercel are both infinitely easier than GH Pages. But there are some downsides:

* Vercel's free plan does not support teams
* Netlify is reported to be costly in general nowadays
* Cloudflare Pages has a better free plan, but they only give you 200 builds per month

And all of those would require setting up a new account, getting someone (Martin?) to add a payment method, etc.

The jj domain thing doesn't apply here, though — we don't want previews to be on the same domain anyway. So jumping between providers is fine; Vercel/Netlify would give us is an autogenerated preview domain and all new previews would land there.